main.c

```diff
@@ -0,0 +1,194 @@
+};
+
+
+psa_status_t atecc608a_hash_sha256(const uint8_t *input, size_t input_size,
+                                   const uint8_t *expected_hash, size_t expected_hash_size)
+{
+    uint8_t actual_hash[ATCA_SHA_DIGEST_SIZE] = {0};
+
+    printf("SHA-256:\n\n");
+    atcab_printbin_label("Input: ", (uint8_t *)input, input_size);
+    atcab_printbin_label("Expected Hash: ", (uint8_t *)expected_hash, expected_hash_size);
+    ASSERT_STATUS(atcab_init(&atca_iface_config), ATCA_SUCCESS);
+    ASSERT_STATUS(atcab_hw_sha2_256(input, input_size, actual_hash), ATCA_SUCCESS);
+    ASSERT_STATUS(atcab_release(), ATCA_SUCCESS);
+    atcab_printbin_label("Actual Hash: ", actual_hash, ATCA_SHA_DIGEST_SIZE);
+    ASSERT_STATUS(memcmp(actual_hash, expected_hash, sizeof(actual_hash)), 0);
+    printf("Success!\n\n");
+
+    return 0;
+}
+
+psa_status_t atecc608a_print_locked_zones()
+{
+    bool locked;
+    printf("--- Device locks information ---\n");
+    ASSERT_STATUS(atcab_init(&atca_iface_config), ATCA_SUCCESS);
+    ASSERT_STATUS(atcab_is_locked(LOCK_ZONE_CONFIG, &locked), ATCA_SUCCESS);
+    printf("  - Config locked: %d\n", locked);
+    ASSERT_STATUS(atcab_is_locked(LOCK_ZONE_DATA, &locked), ATCA_SUCCESS);
+    printf("  - Data locked: %d\n", locked);
+    for(uint8_t i=0; i < 16; i++)
+    {
+        ASSERT_STATUS(atcab_is_slot_locked(i, &locked), ATCA_SUCCESS);
+        printf("  - Slot %d locked: %d\n", i, locked);
+    }
+    ASSERT_STATUS(atcab_release(), ATCA_SUCCESS);
+    printf("--------------------------------\n");
+    return PSA_SUCCESS;
+}
+
+psa_status_t atecc608a_print_serial_number()
+{
+    uint8_t serial[ATCA_SERIAL_NUM_SIZE];
+    size_t buffer_length;
+
+    if(atecc608a_get_serial_number(serial, ATCA_SERIAL_NUM_SIZE,
+                                   &buffer_length) != PSA_SUCCESS)
+    {
+        return PSA_ERROR_HARDWARE_FAILURE;
+    }
+    printf("Serial Number:\n");
+    atcab_printbin_sp(serial, buffer_length);
+    printf("\n");
+    return PSA_SUCCESS;
+}
+
+int main(void)
+{
+    enum {
+        key_type = PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1),
+        keypair_type = PSA_KEY_TYPE_ECC_KEYPAIR(PSA_ECC_CURVE_SECP256R1),
+        key_bits = 256,
+        hash_alg = PSA_ALG_SHA_256,
+        alg = PSA_ALG_ECDSA(hash_alg),
+        sig_size = PSA_ASYMMETRIC_SIGN_OUTPUT_SIZE(key_type, key_bits, alg),
+        pubkey_size = PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(key_bits),
+        hash_size = PSA_HASH_SIZE(hash_alg),
+    };
+    psa_status_t status;
+    psa_key_handle_t verify_handle;
+    uint8_t signature[sig_size];
+    size_t signature_length = 0;
+    const uint8_t hash[hash_size] = {
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
+    };
```

Comment:
Consider re-using `sha256_expected_hash1` for this.

Reply:
I'd rather go for clarity, since this is just an example.